cmd/run.go

```diff
@@ -273,7 +273,13 @@ a commandline interface for interacting with it.`,
 				}
 				precision := 100 * time.Millisecond
 				atT := engine.Executor.GetTime()
-				if endT := engine.Executor.GetEndTime(); endT.Valid {
+				stagesEndT := lib.SumStages(engine.Executor.GetStages())
+				endT := engine.Executor.GetEndTime()
+				//get both times and use whichever is valid and lower
```

Comment:
This comment is kind of unnecessary. If you feel it needs to be here, capitalise the G and put a space after the "//".